File: 17_Math&Geometry/02_Plus_One/main.py
class Solution:
    def plusOne(self, digits):
        n = len(digits)
        
        for i in range(n - 1, -1, -1):
            if digits[i] < 9:
                digits[i] += 1
                return digits
            digits[i] = 0
        
        return [1] + digits

# Test cases
sol = Solution()

digits1 = [1, 2, 3, 4]
result1 = sol.plusOne(digits1)
print(f"Result: {result1}")

digits2 = [9, 9, 9]
result2 = sol.plusOne(digits2)
print(f"Result: {result2}")

File: 17_Math&Geometry/02_Plus_One/main.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> plusOne(std::vector<int>& digits) {
        int n = digits.size();
        
        for (int i = n - 1; i >= 0; --i) {
            if (digits[i] < 9) {
                ++digits[i];
                return digits;
            }
            digits[i] = 0;
        }
        
        digits.insert(digits.begin(), 1);
        return digits;
    }
};

int main() {
    Solution sol;
    std::vector<int> digits1 = {1, 2, 3, 4};
    std::vector<int> result1 = sol.plusOne(digits1);
    std::cout << "Result: ";
    for (int digit : result1) {
        std::cout << digit << " ";
    }
    std::cout << std::endl;

    std::vector<int> digits2 = {9, 9, 9};
    std::vector<int> result2 = sol.plusOne(digits2);
    std::cout << "Result: ";
    for (int digit : result2) {
        std::cout << digit << " ";
    }
    std::cout << std::endl;

    return 0;
}
